src/shared/services/UnreadCounterService.ts

```diff
@@ -0,0 +1,95 @@
+import { UserService, HttpService } from "../services";
+import { updateUnreadCountsInterval } from "../config";
+import { poll } from "@utils/helpers";
+import { myAuth } from "@utils/app";
+import { amAdmin } from "@utils/roles";
+import { isBrowser } from "@utils/browser";
+
+/**
+ * Service to poll and keep track of unread messages / notifications.
+ */
+export class UnreadCounterService {
+  // fetched by HttpService.getUnreadCount, appear in inbox
+  unreadPrivateMessages = 0;
+  unreadReplies = 0;
+  unreadMentions = 0;
+  public unreadInboxCount = 0;
+
+  // fetched by HttpService.getReportCount, appear in report page
+  commentReportCount = 0;
+  postReportCount = 0;
+  messageReportCount = 0;
+  public unreadReportCount = 0;
+
+  // fetched by HttpService.getUnreadRegistrationApplicationCount, appear in registration application page
+  public unreadApplicationCount = 0;
+
+  static #instance: UnreadCounterService;
+  private subscriptions = new Set<(service: UnreadCounterService) => any>();
+  private fetching = false;
+
+  constructor() {
+    if (isBrowser()) {
+      poll(this.update, updateUnreadCountsInterval);
```

Comment:
I am pretty sure this poll function just leaks memory over time due to the recursion (need a new stack frame every time)

Reply:
You could try using an interval, although that may require a change to `poll` that is outside of the context of this PR.